canonical-web-and-design/vanilla-framework
# Use Codepen for Vanilla docs examples

Fixes https://github.com/canonical-web-and-design/example-js/issues/51


## Done

- removed references to previous example-js implementation
- added script to replace example links with CodePen embeds
- cleaned up js-example links markup in docs

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/ or https://vanilla-framework-canonical-web-and-design-pr-2705.run.demo.haus/
- Browse documentation (patterns, utilities), see examples rendered via CodePen
- Make sure examples render correctly

## Details

We are using default dark CodePen theme.

CodePen embed uses hotlinked latest published version of Vanilla CSS.
Additionally has small CSS linked to add 1rem margin around body, to avoid examples sticking to embed borders.

All embeds have default height of 400px (as it may be more than needed for smaller examples, the embed needs to fit the code as well).
It's possible to change the height of individual examples via `data-height` property on the example link (see [Embedded media utility](https://vanilla-framework-canonical-web-and-design-pr-2705.run.demo.haus/utilities/embedded-media/))

Because CodePen embeds rely on external script provided by CodePen if that would fail we will fallback to displaying code of the example in standard Vanilla code blocks.

## Screenshots

Example pattern:
<img width="1090" alt="Screenshot 2019-12-13 at 14 11 16" src="https://user-images.githubusercontent.com/83575/70802732-7e62a680-1db2-11ea-91b5-b63ac047d440.png">

HTML code open with result:
<img width="1073" alt="Screenshot 2019-12-13 at 14 11 25" src="https://user-images.githubusercontent.com/83575/70802731-7e62a680-1db2-11ea-857f-1c7261b9820c.png">

JS open:
<img width="1065" alt="Screenshot 2019-12-13 at 14 11 32" src="https://user-images.githubusercontent.com/83575/70802730-7dca1000-1db2-11ea-8ab2-5d78112ec18c.png">
